packages/wonder-blocks-announcer/src/announcer.ts

```diff
@@ -0,0 +1,257 @@
+        }
+        return Announcer._instance;
+    }
+    /**
+     * Internal initializer method to create live region elements
+     * Prepends regions to document body
+     * @param {string} id ID of the top level node (wbAnnounce)
+     */
+    init(id: string) {
+        this.node = document.createElement("div");
+        this.node.id = id;
+        this.node.setAttribute("data-testid", id);
+
+        Object.assign(this.node.style, srOnly);
+
+        // For each level, we create at least two live region elements.
+        // This is to work around AT occasionally dropping messages.
+        const aWrapper = createRegionWrapper("assertive");
+        createDuplicateRegions(
+            aWrapper,
+            "assertive",
+            this.regionFactory.count,
+            this.dictionary,
+        );
+        this.node?.appendChild(aWrapper);
+
+        const pWrapper = createRegionWrapper("polite");
+        createDuplicateRegions(
+            pWrapper,
+            "polite",
+            this.regionFactory.count,
+            this.dictionary,
+        );
+        this.node.appendChild(pWrapper);
+
+        document.body.append(this.node);
+    }
+    /**
+     * Recover in the event regions get lost
+     * This happens in Storybook or other HMR environments when saving a file:
+     * Announcer exists, but it loses the connection to DOM element Refs
+     */
+    reattachNodes() {
+        const announcerCheck = document.getElementById(`wbAnnounce`);
+        if (announcerCheck !== null) {
+            this.node = announcerCheck;
+            const regions = Array.from(
+                announcerCheck.querySelectorAll<HTMLElement>(
+                    "[id^='wbARegion']",
+                ),
+            );
+            regions.forEach((region) => {
+                this.dictionary.set(region.id, {
+                    id: region.id,
+                    levelIndex: parseInt(
+                        region.id.charAt(region.id.length - 1),
+                    ),
+                    level: region.getAttribute("aria-live") as PolitenessLevel,
+                    element: region,
+                });
+            });
+        }
+    }
+
+    /**
+     * Announce a live region message for a given level
+     * @param {string} message The message to be announced
+     * @param {string} level Politeness level: should it interrupt?
+     * @param {number} debounceThreshold Optional duration to wait before appending another message (defaults to 250ms)
+     * @returns {Promise<string>} Promise that resolves with an IDREF for targeted element or empty string if it failed
+     */
+    async announce(
+        message: string,
+        level: PolitenessLevel,
+        debounceThreshold: number = DEFAULT_WAIT_THRESHOLD,
+    ): Promise<string> {
+        // Locking variable to prevent simultaneous updates
+        if (this.stateLock) {
+            return Promise.resolve(""); // Return early if state is locked
```

Comment:
**question**: If `announce` is called and `stateLock` is true, is there a waiting/re-try mechanism for the message?